Feature 2925 update contributions docs (#2932)
mercedes-benz/sechub: README.adoc

=== Introduction

> SecHub orchestrates various security and vulnerability scanners which can find potential vulnerabilities in the software under test.

Reviewer: ```suggestion
SecHub orchestrates various security and vulnerability scanners which can find potential vulnerabilities in sourcecode, binaries or web applications.
```

Author: changed

                                              +--> | PDS + Tool A |
                                              |    +--------------+
+--------+                     +---------+    |
| SecHub | ---- scan data ---> | SecHub  | <--+
| Client | <---- report ------ |   API   | <--+
+--------+                     +---------+    |
                                              |    +--------------+
                                              +--> | PDS + Tool B |
                                                   +--------------+
....

The objective of SecHub is to help secure the software development lifecyle (SDLC) phases: development, deployment and maintenance. From the first written line of code to the application being in production. SecHub can be used to scan the software continuously.

> To reduce the cognitive load for the user, the security tools are categorized into modules.
> Modules are named after the security testing method they perform: `codeScan`, `licenseScan`, `secretScan`, `webScan` etc.

Reviewer: ```suggestion
The security tools are categorized into modules which are named after the security testing method they perform: `codeScan`, `licenseScan`, `secretScan`, `webScan` etc.
```

Author: changed

[NOTE]
--
The terms SAST (Static Application Security Testing) and DAST (Dynamic Application Security Testing) are intentionally not used for the module names, because the designers feel those terms are vague and difficult to understand for non-security experts. On the other hand, security experts can easily map: `codeScan` to SAST and `webScan` to DAST.
--

Because of the modules, the user only needs to understand what security testing a module performs, rather than to know what specific security tools are used for the actual scan. The user describes in a configuration file what module(s) to use. Or in other words, what security testing methods should be used to test the software with.

> To start a scan, the user sends the configuration file to the SecHub server via API. Even better, SecHub provides a client which calls the REST API for the user. The SecHub client can be used manually or can be integrated into a continues-integration build pipeline (see below figure).

Reviewer: ```suggestion
To start a scan, the user sends the configuration file to the SecHub server via API. For ease of use SecHub offers a client which calls the REST API for the user. The SecHub client can be used manually or can be integrated into a continues-integration build pipeline (see below figure).
```

Author: changed

}
----

The report can be downloaded in two flavors: JSON and HTML. Both are human readable. The HTML report is self-contained and can be read in any browser.
The JSON format is machine readable and can be read by the <<sechub-plugins, SecHub plugins>>.

[[mark-false-positives]]
==== Mark Findings as False-Positives

> There are two major reasons for marking a security finding as false-positive. +
> - First, it is an actual false-positive. +
> - Second, the finding is a false-positive in the context of the application. For example, the application is never deployed to be reachable from the internet.

Reviewer: ```suggestion
There are two major reasons for marking a security finding as false-positive: +
- It is an actual false-positive. +
- The finding is a false-positive in the context of the application. For example, the application is never deployed to be reachable from the internet.
```

Author: changed

Each module performs a different security testing method:

> * <<codescan, `codeScan`>> - scans the code for potential vulnerabilities (weaknesses). Also known as SAST or static source code analysis.

Reviewer: ```suggestion
* <<codescan, `codeScan`>> - scans the code or binaries for potential vulnerabilities (weaknesses). This includes SAST (static application security testing) and IaC (infrastructure as code).
```

Author: changed

The smallest useful setup is: a single SecHub server and a single product delegation server (PDS). Those two components are enough to start scanning.

For a larger setup, the number of SecHub server instances can be increased. More PDS instances can be added. +
The only requirements to scale SecHub are: a PostgreSQL database and an object store or file share.
SecHub and PDS instances use the PostgreSQL database to share information between instances.
For example, the job queue is kept in PostgreSQL. In addition,an object store or file share is necessary, so that all SecHub or PDS instances can store/read files.

Regardless of the backend complexity, whether one SecHub server or many are used the <<user-perspective, workflow for the user>> stays the same.

For more details about the architecture have a look at the architecture documentation: https://mercedes-benz.github.io/sechub/latest/sechub-architecture.html.

=== Operations Perspective

The operations team has to do most of the work.
The operations team of SecHub needs to configure the scan tools as well as manage users and projects.
This is an intentional design choice.

> The idea is, that users like security experts, operations engineers and developers can just scan and therefore focus on their work, rather than take care of SecHub and security scanners.

Reviewer: Maybe we should cut this part out. Users might not split their teams in DEV and OPS and instead have the same people run and use SecHub.

Author: rewrote the section
